packages/framework/src/Model/SocialNetwork/SocialNetworkConfigFactory.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Model\SocialNetwork;
+
+use App\Component\SocialNetwork\Seznam;
+
+class SocialNetworkConfigFactory
+{
+    public const PROVIDER_GOOGLE = 'Google';
+    public const PROVIDER_FACEBOOK = 'Facebook';
+    public const PROVIDER_SEZNAM = 'Seznam';
+
+    /**
+     * @param string $googleClientId
+     * @param string $googleClientSecret
+     * @param string $facebookClientId
+     * @param string $facebookClientSecret
+     * @param string $seznamClientId
+     * @param string $seznamClientSecret
+     */
+    public function __construct(
```

Comment:
If config in yaml, I would rather create a true framework bundle config than use parameters and have to validate them by hand – you can see inspiration in https://github.com/shopsys/shopsys/pull/3084/commits/9636a0fad48b4142c11394db75b0a7712915f58c

Reply:
@grossmannmartin Honestly, solution from Rosta seems to me better because creating "yaml configuration" (or how to name it) is not good with this solution because with configuration you tell what can be used and what is expected will come. This library for login by social network has many solutions and I can't all of them solve + if library whatever change you have to update your configuration ++ if on project you need to do something that is not expected you have to edit this configuration too. Because I don't think creating configuration is not good solution and I accept solution of creating only yaml file where I will send only config as array as Rosta suggest.

Do you agree or do you want to call so I can better describe issue with configuration solution? 🙏

Reply:
changed by discussion and suggestion by Rosta, see [68ae66f](https://github.com/shopsys/shopsys/pull/3154/commits/68ae66fd9ef6fd6cfcdaf20831bd36b8e38f3186). After discussion, I take it's OK now, if not let me know